src/bridge/MatrixHandler.ts

```diff
@@ -538,6 +538,38 @@ export class MatrixHandler {
             }
             // get the virtual IRC user for this user
             promises.push((async () => {
+                const entry = await this.ircBridge.getStore().getRoom(event.room_id, room.server.domain, room.channel);
```

Comment:
This *sorta* feels like it could be it's own function, this async block is already chonky.